File: DynamsoftBarcodeReaderDemo/BarcodeTypesTableViewController.m

#import "BarcodeTypesTableViewController.h"
#import <DynamsoftBarcodeReader/Barcode.h>

@interface BarcodeTypesTableViewController ()

@end

@implementation BarcodeTypesTableViewController

@synthesize barcodeTypesTableView;
@synthesize mainView;

@synthesize linearCell;
@synthesize qrcodeCell;
@synthesize pdf417Cell;
@synthesize datamatrixCell;

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [barcodeTypesTableView setEditing: YES animated: NO];
    
    [self configCellsBackground];
    
    [self selectCells];
    
    barcodeTypesTableView.tableFooterView = [[UIView alloc] initWithFrame:CGRectZero];
}

-(void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    
    if(mainView != nil && mainView.dbrManager != nil)
        mainView.dbrManager.isPauseFramesComing = YES;
}

- (void) configCellsBackground{
    UIView *bgColorView = [[UIView alloc] init];
    bgColorView.backgroundColor = [UIColor whiteColor];
    
    [linearCell setSelectedBackgroundView:bgColorView];
    [qrcodeCell setSelectedBackgroundView:bgColorView];
    [pdf417Cell setSelectedBackgroundView:bgColorView];
    [datamatrixCell setSelectedBackgroundView:bgColorView];
}

- (void)selectCells{
    long types = (mainView == nil || mainView.dbrManager == nil) ? [Barcode UNKNOWN] : mainView.dbrManager.barcodeFormat;
    
    if((types | [Barcode OneD]) == types)
    {
        NSIndexPath *indexPath=[NSIndexPath indexPathForRow:0 inSection:0];
        [barcodeTypesTableView selectRowAtIndexPath:indexPath animated:NO  scrollPosition:UITableViewScrollPositionBottom];
    }

    if((types | [Barcode QR_CODE]) == types)
    {
        NSIndexPath *indexPath=[NSIndexPath indexPathForRow:1 inSection:0];
        [barcodeTypesTableView selectRowAtIndexPath:indexPath animated:NO  scrollPosition:UITableViewScrollPositionBottom];
    }
    
    if((types | [Barcode PDF417]) == types)
    {
        NSIndexPath *indexPath=[NSIndexPath indexPathForRow:2 inSection:0];
        [barcodeTypesTableView selectRowAtIndexPath:indexPath animated:NO  scrollPosition:UITableViewScrollPositionBottom];
    }
    
    if((types | [Barcode DATAMATRIX]) == types)
    {
        NSIndexPath *indexPath=[NSIndexPath indexPathForRow:3 inSection:0];
        [barcodeTypesTableView selectRowAtIndexPath:indexPath animated:NO  scrollPosition:UITableViewScrollPositionBottom];
    }
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    //where indexPath.row is the selected cell
}

- (void)tableView:(UITableView *)tableView didDeselectRowAtIndexPath:(NSIndexPath *)indexPath {
    //where indexPath.row is the selected cell
    BOOL hasCellSelected = linearCell.selected || qrcodeCell.selected || pdf417Cell.selected || datamatrixCell.selected;
    
    if(hasCellSelected == NO)
        [barcodeTypesTableView selectRowAtIndexPath:indexPath animated:NO  scrollPosition:UITableViewScrollPositionBottom];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (UITableViewCellEditingStyle)tableView:(UITableView*)tableView editingStyleForRowAtIndexPath:(NSIndexPath*)indexPath
{
    return UITableViewCellEditingStyleDelete | UITableViewCellEditingStyleInsert;
}

-(void) viewWillDisappear:(BOOL)animated {
    if ([self.navigationController.viewControllers indexOfObject:self]==NSNotFound) {
        long types = 0;
        if(linearCell.selected)
            types |= [Barcode OneD];
        if(qrcodeCell.selected)
            types |= [Barcode QR_CODE];
        if(pdf417Cell.selected)
            types |= [Barcode PDF417];
        if(datamatrixCell.selected)
            types |= [Barcode DATAMATRIX];
        
        if(mainView != nil && mainView.dbrManager != nil)
            mainView.dbrManager.barcodeFormat = types;
    }
    [super viewWillDisappear:animated];
}

@end

File: iOS_DynamsoftBarcodeReaderDemo_swift/DynamsoftBarcodeReaderDemo/BarcodeTypesTableViewController.swift
//
//  BarcodeTypesTableViewController.swift
//  DynamsoftBarcodeReaderDemo
//
//  Created by Dynamsoft on 08/07/2018.
//  Copyright © 2018 Dynamsoft. All rights reserved.
//

import UIKit

class BarcodeTypesTableViewController: UITableViewController {
    
    var mainView: ScanViewController!
    @IBOutlet var barcodeTypesTableView: UITableView!
    @IBOutlet weak var linearCell: UITableViewCell!
    @IBOutlet weak var qrcodeCell: UITableViewCell!
    @IBOutlet weak var pdf417Cell: UITableViewCell!
    @IBOutlet weak var datamatrixCell: UITableViewCell!
    
    @IBOutlet weak var aztecCell: UITableViewCell!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        barcodeTypesTableView.setEditing(true, animated: false);
        self.configCellsBackground();
        self.selectCells();
        barcodeTypesTableView.tableFooterView = UIView.init(frame: CGRect.zero);
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated);
        if(mainView != nil && mainView!.dbrManager != nil)
        {
            mainView.dbrManager?.isPauseFramesComing = true;
        }
    }
    
    func configCellsBackground(){
        let bgColorView = UIView.init();
        bgColorView.backgroundColor = UIColor.white;
        linearCell.selectedBackgroundView = bgColorView;
        qrcodeCell.selectedBackgroundView = bgColorView;
        pdf417Cell.selectedBackgroundView = bgColorView;
        datamatrixCell.selectedBackgroundView = bgColorView;
        aztecCell.selectedBackgroundView = bgColorView;
    }
    
    func selectCells(){
        let types = (mainView == nil || mainView.dbrManager == nil) ? BarcodeType.ALL.rawValue: mainView.dbrManager?.barcodeFormat;
        if((types! | BarcodeType.ONED.rawValue) == types)
        {
            let indexPath = IndexPath(row: 0, section: 0);
            barcodeTypesTableView.selectRow(at: indexPath, animated: false, scrollPosition: UITableViewScrollPosition.bottom);
        }
        if((types! | BarcodeType.QRCODE.rawValue) == types)
        {
            let indexPath = IndexPath(row: 1, section: 0);
            barcodeTypesTableView.selectRow(at: indexPath, animated: false, scrollPosition: UITableViewScrollPosition.bottom);
        }
        if((types! | BarcodeType.PDF417.rawValue) == types)
        {
            let indexPath = IndexPath(row: 2, section: 0);
            barcodeTypesTableView.selectRow(at: indexPath, animated: false, scrollPosition: UITableViewScrollPosition.bottom);
        }
        if((types! | BarcodeType.DATAMATRIX.rawValue) == types)
        {
            let indexPath = IndexPath(row: 3, section: 0);
            barcodeTypesTableView.selectRow(at: indexPath, animated: false, scrollPosition: UITableViewScrollPosition.bottom);
        }
        if((types! | BarcodeType.AZTEC.rawValue) == types)
        {
            let indexPath = IndexPath(row: 4, section: 0);
            barcodeTypesTableView.selectRow(at: indexPath, animated: false, scrollPosition: UITableViewScrollPosition.bottom);
        }
    }
    
    override func tableView(_ tableView: UITableView, didDeselectRowAt indexPath: IndexPath) {
        //where indexPath.row is the selected cell
        let hasCellSelected = linearCell.isSelected || qrcodeCell.isSelected || pdf417Cell.isSelected || datamatrixCell.isSelected ||
            aztecCell.isSelected;
        if(hasCellSelected == false)
        {
            barcodeTypesTableView.selectRow(at: indexPath, animated: false, scrollPosition: UITableViewScrollPosition.bottom);
        }
    }
    
    override func tableView(_ tableView: UITableView, editingStyleForRowAt indexPath: IndexPath) -> UITableViewCellEditingStyle {
        return UITableViewCellEditingStyle.init(rawValue: UITableViewCellEditingStyle.insert.rawValue | UITableViewCellEditingStyle.delete.rawValue)!;
    }

    override func viewWillDisappear(_ animated: Bool) {
        var types = 0;
        if(linearCell.isSelected)
        {
            types = types | BarcodeType.ONED.rawValue;
        }
        if(qrcodeCell.isSelected)
        {
            types = types | BarcodeType.QRCODE.rawValue;
        }
        if(pdf417Cell.isSelected)
        {
            types = types | BarcodeType.PDF417.rawValue;
        }
        if(datamatrixCell.isSelected)
        {
            types = types | BarcodeType.DATAMATRIX.rawValue;
        }
        if(aztecCell.isSelected)
        {
            types = types | BarcodeType.AZTEC.rawValue;
        }
        if(mainView != nil && mainView!.dbrManager != nil)
        {
            mainView.dbrManager?.setBarcodeFormat(format: types);
        }
        super.viewWillDisappear(animated);
    }
    
}
